aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -65,4 +236,148 @@ Tensor _s_where_cpu(const Tensor& condition, const Tensor& self, const Tensor& o
+}
+
+Tensor gt_cpu(const Tensor& self, const Tensor& other) {
+  return cmp_cpu<std::greater>(self, other, "gt");
+}
+
+Tensor le_cpu(const Tensor& self, const Tensor& other) {
+  return cmp_cpu<std::less_equal>(self, other, "le");
+}
+
+Tensor ge_cpu(const Tensor& self, const Tensor& other) {
+  return cmp_cpu<std::greater_equal>(self, other, "le");
+}
+
+Tensor eq_cpu(const Tensor& self, const Tensor& other) {
+  return cmp_cpu<std::equal_to>(self, other, "eq");
+}
+
+Tensor ne_cpu(const Tensor& self, const Tensor& other) {
+  return cmp_cpu<std::not_equal_to>(self, other, "ne");
+}
+
+Tensor& lt_out_cpu(Tensor& result, const Tensor& self, const Tensor& other) {
+  return cmp_out_cpu<std::less>(result, self, other, "lt");
+}
+
+Tensor& gt_out_cpu(Tensor& result, const Tensor& self, const Tensor& other) {
+  return cmp_out_cpu<std::greater>(result, self, other, "gt");
+}
+
+Tensor& le_out_cpu(Tensor& result, const Tensor& self, const Tensor& other) {
+  return cmp_out_cpu<std::less_equal>(result, self, other, "le");
+}
+
+Tensor& ge_out_cpu(Tensor& result, const Tensor& self, const Tensor& other) {
+  return cmp_out_cpu<std::greater_equal>(result, self, other, "ge");
+}
+
+Tensor& eq_out_cpu(Tensor& result, const Tensor& self, const Tensor& other) {
+  return cmp_out_cpu<std::equal_to>(result, self, other, "eq");
+}
+
+Tensor& ne_out_cpu(Tensor& result, const Tensor& self, const Tensor& other) {
+  return cmp_out_cpu<std::not_equal_to>(result, self, other, "ne");
+}
+
+Tensor& lt_inplace_cpu(Tensor& self, Scalar other) {
+  return cmp_inplace_cpu<std::less>(self, other, "lt_");
+}
+
+Tensor& gt_inplace_cpu(Tensor& self, Scalar other) {
+  return cmp_inplace_cpu<std::greater>(self, other, "gt_");
+}
+
+Tensor& le_inplace_cpu(Tensor& self, Scalar other) {
+  return cmp_inplace_cpu<std::less_equal>(self, other, "le_");
+}
+
+Tensor& ge_inplace_cpu(Tensor& self, Scalar other) {
+  return cmp_inplace_cpu<std::greater_equal>(self, other, "ge_");
+}
+
+Tensor& eq_inplace_cpu(Tensor& self, Scalar other) {
+  return cmp_inplace_cpu<std::equal_to>(self, other, "eq_");
+}
+
+Tensor& ne_inplace_cpu(Tensor& self, Scalar other) {
+  return cmp_inplace_cpu<std::not_equal_to>(self, other, "ne_");
+}
+
+Tensor& lt_inplace_cpu(Tensor& self, const Tensor& other) {
+  return cmp_inplace_cpu<std::less>(self, other, "lt_");
+}
+
+Tensor& gt_inplace_cpu(Tensor& self, const Tensor& other) {
+  return cmp_inplace_cpu<std::greater>(self, other, "gt_");
+}
+
+Tensor& le_inplace_cpu(Tensor& self, const Tensor& other) {
+  return cmp_inplace_cpu<std::less_equal>(self, other, "le_");
+}
+
+Tensor& ge_inplace_cpu(Tensor& self, const Tensor& other) {
+  return cmp_inplace_cpu<std::greater_equal>(self, other, "ge_");
+}
+
+Tensor& eq_inplace_cpu(Tensor& self, const Tensor& other) {
+  return cmp_inplace_cpu<std::equal_to>(self, other, "eq_");
+}
+
+Tensor& ne_inplace_cpu(Tensor& self, const Tensor& other) {
+  return cmp_inplace_cpu<std::not_equal_to>(self, other, "ne_");
+}
```

Comment:
Wow, that's quite a lot of functions. It looks like they only differ by comparators and names, so can we implement them using macros? We usually define a single case and then unroll it with another `FORALL` macro that instantiates it for all valid combinations. You can grep for it in ATen and you should find some examples